### ISSUE TITLE
Update to Docker Compose 1.23.0

### ISSUE THREAD

**Issue by sean-e-dietrich** [docksal/docksal#819]
**Update:**

Docker Compose v1.23.0 has changed it's naming scheme which breaks existing assumptions about container names in a Docker Compose managed stack (reported in docker/compose#6316)

---

Database commands rely on the container name being `${COMPOSE_PROJECT_NAME_SAFE}_db_1`

There have been times where docker compose will append a hash to the end of the container ID and therefore those containers are no longer usable with the `fin db ...` commands.

The following was a conversation in Drupal Slack 

```
fabean [12:38 PM]
@sean.e.dietrich different weird issue. Now (I assume unrelated) all my containers are `projectname_cli_1-HASH` which is breaking `fin` commands from working with those containers.

2018-11-13-144005_grim.png 

This is on linux. I tried `fin system stop` then `docker system prune` I've restarted docker, and my computer. I also tried spinning up a new site and get the same sort of things on the end of my container names

sean.e.dietrich [12:39 PM]
can you run `fin --version`

fabean [12:40 PM]
1.69.1

sean.e.dietrich [12:40 PM]
what happens when you run `fin restart`

fabean [12:41 PM]
@sean.e.dietrich it restarts those project containers with the hashes still on the end

sean.e.dietrich [12:42 PM]
you mentioned that `fin` is not working with the hashes correct at the end of the containers?

fabean [12:42 PM]
yes, for example running `fin db create databasename`
says `Error: No such container: scce-drupal_db_1`

sean.e.dietrich [12:44 PM]
ahh ok
are you at a point you can destroy the db container and rebuild it?

fabean [12:45 PM]
Yes I can delete everything no problem

sean.e.dietrich [12:45 PM]
if you run `fin reset db` that should rebuild the database container

fabean [12:48 PM]
Yes it does, same issue of having hashes on the end of my containers though.
```

![2018-11-13-144005_grim](https://user-images.githubusercontent.com/1564748/48442221-6c8f7980-e742-11e8-82ca-aa9c5ab78648.png)

**Comment by sean-e-dietrich** [comment]
![2018-11-13-145233_grim](https://user-images.githubusercontent.com/1564748/48442697-c9d7fa80-e743-11e8-931a-40537f50ad3e.png)

**Comment by fabean** [comment]
`fin sysinfo`

███  OS
Linux Arch Linux 
Linux Golduck 4.18.16-arch1-1-macbook #1 SMP PREEMPT Sat Oct 27 14:03:01 EDT 2018 x86_64 GNU/Linux

███  ENVIRONMENT
MODE : Linux Kernel
DOCKER_HOST : 

███  FIN
fin version: 1.69.1

███  DOCKER COMPOSE
EXPECTED VERSION: 1.21.1
docker-compose version 1.23.1, build unknown
docker-py version: 3.5.1
CPython version: 3.7.1
OpenSSL version: OpenSSL 1.1.1  11 Sep 2018

███  DOCKER
EXPECTED VERSION: 18.03.1-ce

Client:
 Version:           18.09.0-ce
 API version:       1.39
 Go version:        go1.11.2
 Git commit:        4d60db472b
 Built:             Fri Nov  9 00:05:34 2018
 OS/Arch:           linux/amd64
 Experimental:      false

Server:
 Engine:
  Version:          18.09.0-ce
  API version:      1.39 (minimum version 1.12)
  Go version:       go1.11.2
  Git commit:       4d60db472b
  Built:            Fri Nov  9 00:05:11 2018
  OS/Arch:          linux/amd64
  Experimental:     false

███  DOCKSAL: PROJECTS
project             STATUS              virtual host                                project root
scce-drupal         Up 8 minutes        scce-drupal.docksal,*.scce-drupal.docksal   /home/joshfabean/Sites/scce-drupal

███  DOCKSAL: VIRTUAL HOSTS
[0;32mhttp://*.scce-drupal.docksal[0m
[0;32mhttp://scce-drupal.docksal[0m

███  DOCKER: RUNNING CONTAINERS
CONTAINER ID        IMAGE                       COMMAND                  CREATED             STATUS                   PORTS                                                    NAMES
c8d4b78207d1        docksal/web:2.1-apache2.4   "httpd-foreground"       8 minutes ago       Up 8 minutes             80/tcp, 443/tcp                                          scce-drupal_web_1_24d555a9ac04
163e09303ea7        docksal/cli:2.4-php7.2      "/opt/startup.sh sup…"   8 minutes ago       Up 8 minutes (healthy)   22/tcp, 3000/tcp, 9000/tcp                               scce-drupal_cli_1_95dfa2baeac8
b320680d71db        docksal/db:1.1-mysql-5.6    "/entrypoint.sh mysq…"   8 minutes ago       Up 8 minutes             0.0.0.0:32775->3306/tcp                                  scce-drupal_db_1_8e5258ee0e71
d1e8c614f828        docksal/ssh-agent:1.0       "/run.sh ssh-agent"      18 minutes ago      Up 18 minutes                                                                     docksal-ssh-agent
edb2171bf1ec        docksal/dns:1.0             "/opt/entrypoint.sh …"   18 minutes ago      Up 18 minutes            192.168.64.100:53->53/udp                                docksal-dns
781d6ebb382f        docksal/vhost-proxy:1.3     "docker-entrypoint.s…"   18 minutes ago      Up 18 minutes            192.168.64.100:80->80/tcp, 192.168.64.100:443->443/tcp   docksal-vhost-proxy

███  DOCKER: NETWORKS
NETWORK ID          NAME                       DRIVER              SCOPE
cbf128bfa19b        _default                   bridge              local
961b138c307d        bridge                     bridge              local
17e5b60d33d5        clockwork-drupal_default   bridge              local
42b6d9899961        crowncentersite_default    bridge              local
c5d1019cc4d0        host                       host                local
8f46be4003cf        none                       null                local
96c064a251b1        scce-drupal_default        bridge              local

███  HDD Usage
Filesystem      Size  Used Avail Use% Mounted on
dev             7.8G     0  7.8G   0% /dev
run             7.8G  1.6M  7.8G   1% /run
/dev/sda5       274G  191G   70G  74% /
tmpfs           7.8G  494M  7.3G   7% /dev/shm
tmpfs           7.8G     0  7.8G   0% /sys/fs/cgroup
tmpfs           7.8G  7.5M  7.8G   1% /tmp
/dev/sda1       197M  101M   97M  51% /boot
tmpfs           1.6G   35M  1.6G   3% /run/user/1000


`fin config`
---------------------
COMPOSE_PROJECT_NAME_SAFE: scce-drupal
COMPOSE_FILE:
/home/joshfabean/.docksal/stacks/volumes-bind.yml
/home/joshfabean/.docksal/stacks/stack-default.yml
/home/joshfabean/Sites/scce-drupal/.docksal/docksal.yml
ENV_FILE:
/home/joshfabean/Sites/scce-drupal/.docksal/docksal.env

PROJECT_ROOT: /home/joshfabean/Sites/scce-drupal
DOCROOT: docroot
VIRTUAL_HOST: scce-drupal.docksal
VIRTUAL_HOST_ALIASES: *.scce-drupal.docksal
IP: 192.168.64.100
MYSQL: 192.168.64.100:32775

Docker Compose configuration
---------------------
services:
  cli:
    dns:
    - 192.168.64.100
    - 208.67.222.222
    - 192.168.64.100
    - 208.67.222.222
    - 192.168.64.100
    - 208.67.222.222
    environment:
      BASE_URL: null
      DATABASE_HOST: db
      DATABASE_NAME: default
      DATABASE_PASSWORD: user
      DATABASE_USERNAME: user
      DOCROOT: docroot
      HOST_GID: '33'
      HOST_UID: '1000'
      SECRET_ACAPI_EMAIL: null
      SECRET_ACAPI_KEY: null
      SECRET_BLACKFIRE_CLIENT_ID: null
      SECRET_BLACKFIRE_CLIENT_TOKEN: null
      SECRET_PLATFORMSH_CLI_TOKEN: null
      SECRET_SSH_PRIVATE_KEY: null
      SECRET_TERMINUS_TOKEN: null
      VIRTUAL_HOST: scce-drupal.docksal
      XDEBUG_CONFIG: remote_connect_back=0 remote_host=192.168.64.1
      XDEBUG_ENABLED: '1'
    hostname: cli
    image: docksal/cli:2.4-php7.2
    volumes:
    - docksal_ssh_agent:/.ssh-agent:ro
    - cli_home:/home/docker:rw
    - project_root:/var/www:rw,nocopy
  db:
    dns:
    - 192.168.64.100
    - 208.67.222.222
    - 192.168.64.100
    - 208.67.222.222
    - 192.168.64.100
    - 208.67.222.222
    environment:
      MYSQL_ALLOW_EMPTY_PASSWORD: null
      MYSQL_DATABASE: default
      MYSQL_INITDB_SKIP_TZINFO: null
      MYSQL_ONETIME_PASSWORD: null
      MYSQL_PASSWORD: user
      MYSQL_RANDOM_ROOT_PASSWORD: null
      MYSQL_ROOT_PASSWORD: root
      MYSQL_USER: user
    hostname: db
    image: docksal/db:1.1-mysql-5.6
    ports:
    - 3306/tcp
    volumes:
    - /home/joshfabean/Sites/scce-drupal/db:/docker-entrypoint-initdb.d:ro
    - db_data:/var/lib/mysql:rw
    - project_root:/var/www:ro,nocopy
  web:
    depends_on:
      cli:
        condition: service_started
    dns:
    - 192.168.64.100
    - 208.67.222.222
    - 192.168.64.100
    - 208.67.222.222
    - 192.168.64.100
    - 208.67.222.222
    environment:
      APACHE_BASIC_AUTH_PASS: null
      APACHE_BASIC_AUTH_USER: null
      APACHE_DOCUMENTROOT: /var/www/docroot
    hostname: web
    image: docksal/web:2.1-apache2.4
    labels:
      io.docksal.cert-name: none
      io.docksal.project-root: /home/joshfabean/Sites/scce-drupal
      io.docksal.virtual-host: scce-drupal.docksal,*.scce-drupal.docksal
    volumes:
    - project_root:/var/www:ro,nocopy
version: '2.1'
volumes:
  cli_home: {}
  db_data: {}
  docksal_ssh_agent:
    external: true
    name: docksal_ssh_agent
  project_root:
    driver: local
    driver_opts:
      device: /home/joshfabean/Sites/scce-drupal
      o: bind
      type: none

---------------------

**Comment by sean-e-dietrich** [comment]
Looks like Docker Compose v1.23.1 has changed it's naming scheme as reported here https://github.com/docker/compose/issues/6316 

> The default naming scheme for containers created by Compose in this version
has changed from <project>_<service>_<index> to
<project>_<service>_<index>_<slug>

**Comment by lmakarov** [comment]
> Looks like Docker Compose v1.23.1 has changed it's naming scheme

That's a bummer, as we will no longer be able to use the `${COMPOSE_PROJECT_NAME_SAFE}_<service>_1` shortcut and have to query the container name, which is an extra call to docker/docker-compose.

For now, we are still using Docker Compose v1.22.0, but will eventually upgrade to the latest one, which is going to cause issues and require changes.

**Comment by lmakarov** [comment]
@fabean in you case the expected and actual versions of Docker and Docker Compose do not match:

```
███  DOCKER COMPOSE
EXPECTED VERSION: 1.21.1
docker-compose version 1.23.1, build unknown
```
```
███  DOCKER
EXPECTED VERSION: 18.03.1-ce

Client:
 Version:           18.09.0-ce
...
Server:
 Engine:
  Version:          18.09.0-ce
```

You'd have to manually downgrade Docker Compose for now, until we resolve the compatibility issues with the newer versions.

**Comment by achekulaev** [comment]
If I read the python code correctly we might be able to override the slug from the containers for the service.
https://github.com/docker/compose/commit/9b12f489aab49cfb5d436f207a07d70dec745367

**Comment by fabean** [comment]
@lmakarov I was able to grab the older version of docker-compose and everything is working as expected again.

**Comment by lmakarov** [comment]
Resolved via #843 and #853

**Comment by lmakarov** [comment]
FWIW Wodby uses `container_name` in their Docker4Drupal stack and thus does not run into this issue. See https://github.com/wodby/docker4drupal/blob/master/docker-compose.yml

cc @achekulaev
